test/dhtest.c

```diff
@@ -580,19 +491,23 @@ static int run_rfc5114_tests(void)
     BN_free(priv_key);
     OPENSSL_free(Z1);
     OPENSSL_free(Z2);
-
-    fprintf(stderr, "Initialisation error RFC5114 set %d\n", i + 1);
-    ERR_print_errors_fp(stderr);
+    TEST_error("Initialisation error RFC5114 set %d\n", i + 1);
```

Comment:
Another possibility is, of course, to make sure `TEST_error` always calls `ERR_print_errors_fp(stderr)`...